### ISSUE TITLE
Match statement marked unreachable when it shouldn't be

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3303]
```python
def func(e) -> int:
    match e:
        case _ if True:
            return 0
```

![Screen Shot 2022-04-06 at 12 18 33 PM](https://user-images.githubusercontent.com/7040122/162051980-2c5f3f3f-89b5-4034-8c16-5d44b0563ee1.png)

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.236, which I just published. It will also be included in the next release of pylance.
